addons/events/fnc_weaponEvents.sqf

```diff
@@ -0,0 +1,80 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_weaponEvents
+
+Description:
+    Execute weapon events framework.
+
+    class MyWeapon: MyWeapon_base {
+        class EventHandlers {
+            fired = "_this call CBA_fnc_weaponEvents";
+        };
+    };
+
+Parameters:
+    _unit   - Unit that fired the weapon <OBJECT>
+    _weapon - The weapon fired by the unit <STRING>
+
+Returns:
+    Nothing
+
+Examples:
+    (begin example)
+        fired = "_this call CBA_fnc_weaponEvents";
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+params ["_unit", "_weapon"];
+
+private _config = configFile >> "CfgWeapons" >> _weapon >> "CBA_WeaponEvents";
+
+private _isEmpty = _unit ammo _weapon == 0;
+private _onEmpty = true;
+
+if (isNumber (_config >> "onEmpty")) then {
+    _onEmpty = getNumber (_config >> "onEmpty") == 1;
+};
+
+private _fnc_soundSource = {
+    private _soundSourceName = format [QGVAR(soundSource_%1), _soundLocation];
+    private _soundSource = _unit getVariable [_soundSourceName, objNull];
+
+    if !(_soundSource in attachedObjects _unit) then {
+        _soundSource = "Building" createVehicleLocal [0,0,0];
+        _soundSource attachTo [_unit, [0,0,0], _soundLocation];
+        _unit setVariable [_soundSourceName, _soundSource];
+    };
+
+    _soundSource
+};
+
+if (isEmpty) then {
+    private _sound = getText (_config >> "soundEmpty");
+    private _soundLocation = getText (_config >> "soundLocationEmpty");
+
+    if (_sound != "") then {
+        (call _fnc_soundSource) say3D _sound;
+    };
+};
+
+if (!_isEmpty || _onEmpty) then {
+    private _handAction = getText (_config >> "handAction");
+    private _sound = getText (_config >> "sound");
+    private _soundLocation = getText (_config >> "soundLocation");
+    private _delay = getNumber (_config >> "delay");
+
+    [{
+        params ["_unit", "_handAction", "_sound", "_soundSource"];
+
+        if (local _unit) then {
+            _unit playAction _handAction;
+        };
+
+        if (_sound != "") then {
+            _soundSource say3D _sound;
```

Comment:
the sound object dont get delete anywhere so they would stack up. or did i miss something?

Reply:
No point in deleting it, if you can recycle it instead.

Reply:
See L45

Reply:
aha i oversaw that sorry :D

Reply:
but maybe delete it when the unit dies? because else when some units would use that and die they would eather way stack up.

Reply:
Question is if the gain would be worth the effort. I have my doubts.

Reply:
i think it would. else you could take a look at what i did within the ammo cookoff. Object Pooling would be the perfect thing for this i think.

Reply:
I changed it to reuse the sound source from the corpse. But I still don't want to delete them after use.